src/librustc_mir/util/elaborate_drops.rs

```diff
@@ -768,7 +684,99 @@ impl<'l, 'b, 'tcx, D> DropCtxt<'l, 'b, 'tcx, D>
 
         // FIXME(#34708): handle partially-dropped array/slice elements.
         let reset_block = self.drop_flag_reset_block(DropFlagMode::Deep, drop_block, unwind);
-        self.drop_flag_test_block(reset_block, succ, unwind)
+        self.drop_flag_test_block(reset_block, self.succ, unwind)
+    }
+
+    /// Create a loop that drops an array:
```

Comment:
please keep this method where it was before in the source. Otherwise it is significantly harder to review